Include/cpython/longintrepr.h

```diff
@@ -124,6 +124,12 @@ _PyLong_IsCompact(const PyLongObject* op) {
     return op->long_value.lv_tag < (2 << _PyLong_NON_SIZE_BITS);
 }
 
+static inline int
+PyLong_CheckCompact(const PyObject *op)
+{
+    return PyLong_CheckExact(op) && _PyLong_IsCompact((PyLongObject *)op);
```

Comment:
Seems like the cast here should be `(const PyLongObject *)` instead due to some strict compilers. Let me put up a PR.